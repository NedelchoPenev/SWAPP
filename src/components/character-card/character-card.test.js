import React from 'react';
import { shallow, mount } from 'enzyme';

import CharacterCard from './character-card.component'

describe('CharacterCard component', () => {
  let wrapper;
  const imageUrl = 'www.testImage.com';
  const mockName = 'Obi-Wan Kenobi';
  const mockId = 'people.10';
  let mockHistory;
  let mockProps;

  beforeEach(() => {
    mockHistory = { push: jest.fn() };

    mockProps = {
      character: {
        id: mockId,
        name: mockName,
        image: imageUrl
      },
      history: mockHistory
    };

    wrapper = shallow(<CharacterCard.WrappedComponent {...mockProps}/>);
  });

  it('should render CharacterCard component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call history.push with the right string when CharacterCard clicked', () => {
    wrapper.simulate('click');

    expect(mockHistory.push).toHaveBeenCalledWith(
      `/characters/${mockId}`
    );
  });

  it('should render imageUrl', () => {
    expect(wrapper.find("img").prop("src")).toBe(imageUrl);
  });

  it('should render name', () => {
    expect(wrapper.find('h2').text()).toBe(mockName);
  });

  it('should render imageUrl if image is available', () => {
    const newWrapper = mount(<CharacterCard.WrappedComponent {...mockProps}/>)

    expect(newWrapper.props().character.image).toBe(imageUrl);
  });

  it('should render MISSING_IMG if image is not available', () => {
    mockProps = {
      character: {
        id: mockId,
        name: mockName,
        image: null
      },
      history: mockHistory
    };

    const newWrapper = mount(<CharacterCard.WrappedComponent {...mockProps}/>)

    expect(newWrapper.props().character.image).toBe(null);
  });
});