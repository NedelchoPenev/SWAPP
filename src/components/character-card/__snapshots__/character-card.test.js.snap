// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterCard component should render CharacterCard component 1`] = `
<styled.div
  onClick={[Function]}
>
  <img
    alt="Obi-Wan Kenobi"
    src="www.testImage.com"
  />
  <h2>
    Obi-Wan Kenobi
  </h2>
</styled.div>
`;
